subsys/bluetooth/host/ecc.c

```diff
@@ -39,6 +42,34 @@ bool bt_pub_key_is_debug(uint8_t *cmp_pub_key)
 	return memcmp(cmp_pub_key, debug_public_key, BT_PUB_KEY_LEN) == 0;
 }
 
+bool bt_pub_key_is_valid(const uint8_t *key)
+{
+	uint8_t key_be[BT_PUB_KEY_LEN + 1];
+	psa_key_attributes_t attr = PSA_KEY_ATTRIBUTES_INIT;
+	psa_status_t ret;
+	psa_key_id_t handle;
+
+	psa_set_key_type(&attr, PSA_KEY_TYPE_ECC_PUBLIC_KEY(PSA_ECC_FAMILY_SECP_R1));
+	psa_set_key_bits(&attr, 256);
+	psa_set_key_usage_flags(&attr, PSA_KEY_USAGE_DERIVE);
+	psa_set_key_algorithm(&attr, PSA_ALG_ECDH);
+
+	key_be[0] = 0x04;
+	sys_memcpy_swap(&key_be[1], key, BT_PUB_KEY_COORD_LEN);
+	sys_memcpy_swap(&key_be[1 + BT_PUB_KEY_COORD_LEN], &key[BT_PUB_KEY_COORD_LEN],
+			BT_PUB_KEY_COORD_LEN);
```

Comment:
As someone who isn't 100% familiar with this, what does `0x04` refer to? 

It seems that `key_be` contains both the public key, and some 1 octet header

Reply:
https://github.com/zephyrproject-rtos/zephyr/blob/4488ed1a20a97007d451d2ac1f093c413c56e77a/subsys/bluetooth/host/hci_ecc.c#L124-L128

I didn't feel like duplicating that code comment, but I could add something short if you insist :)

Reply:
And:
https://github.com/zephyrproject-rtos/zephyr/blob/4488ed1a20a97007d451d2ac1f093c413c56e77a/subsys/bluetooth/host/hci_ecc.c#L195-L198

Reply:
I added a one line comment